File: quickstart/src/org.ts
import Nile from "@theniledev/js";

var emoji = require('node-emoji');

import * as dotenv from 'dotenv';

dotenv.config({ override: true })

// Get Nile URL and workspace
if (!process.env.NILE_URL || !process.env.NILE_WORKSPACE) {
  console.error(emoji.get('x'), `Error: missing variable NILE_URL or NILE_WORKSPACE in .env .  See .env.defaults for more info and copy it to .env with your values`);
  process.exit(1);
}
const NILE_URL = process.env.NILE_URL!;
const NILE_WORKSPACE = process.env.NILE_WORKSPACE!;
const nile!;

// Set entity
const NILE_ENTITY_NAME = "DB";

async function setupOrg() {

  // Set Nile
  nile = Nile({
    basePath: NILE_URL,
    workspace: NILE_WORKSPACE,
  });

  // login as user
  let email = "shaun@colton.demo"
  let password = "password"
  await nile.users.loginUser({
    loginInfo: {
      email: email,
      password: password,
    },
    }).catch((error:any) => {
      console.error(emoji.get('x'), `Error: Failed to login to Nile as user ${email}: ` + error.message);
      process.exit(1);
    });
  nile.authToken = nile.users.authToken;
  console.log("\n" + emoji.get('arrow_right'), ` Logged into Nile as user ${email}`);
  console.log(`export NILE_ACCESS_TOKEN=${nile.authToken}`);

  // Check if org exists, create if not
  let orgName = "Colton Labs"
  var myOrgs = await nile.organizations.listOrganizations();
  var maybeOrg = myOrgs.find( org => org.name == orgName);
  if (maybeOrg) {
    console.log(emoji.get('dart'), "Org " + orgName + " exists with id " + maybeOrg.id);
  } else {
    await nile.organizations.createOrganization({"createOrganizationRequest" :
    {
      name: orgName,
    }}).then ( (org) => {
      if (org !== null) {
        let orgID = org.id;
        console.log(emoji.get('white_check_mark'), "Created new org " + org.name + " with orgID " + orgID);
      }
    }).catch((error:any) => {
      if (error.message == "org already exists") {
        console.log("Org with name " + orgName + " already exists but cannot get ID");
        process.exit(1);
      } else {
        console.error(error);
        process.exit(1);
      }
    })
  }

}

setupOrg();

File: quickstart-python/src/org.py
#! /usr/bin/env python

import os
import sys

from dotenv import load_dotenv
from emoji import emojize

from nile_api import AuthenticatedClient, Client
from nile_api.api.users import login_user
from nile_api.api.organizations import create_organization, list_organizations
from nile_api.models.login_info import LoginInfo
from nile_api.models.create_organization_request import CreateOrganizationRequest

GOOD = emojize(":check_mark_button:")
BAD = emojize(":red_circle:")
DART = emojize(":dart:", language="alias")
ARROW_RIGHT = emojize(":arrow_right:", language="alias")

load_dotenv(override=True)
params = {
    param: os.environ.get(param)
    for param in ["NILE_URL", "NILE_WORKSPACE"]
}
NILE_ENTITY_NAME = "DB"

for name, value in params.items():
    if not value:
        print(
            f"{BAD} Error: missing environment variable {name}. See .env.defaults for more info and copy it to .env with your values",  # noqa: E501
        )
        sys.exit(1)


def run():
    email = "shaun@colton.demo"
    password = "password"

    token = login_user.sync(
        client=Client(base_url=params["NILE_URL"]),
        workspace=params["NILE_WORKSPACE"],
        json_body=LoginInfo(email=email, password=password),
    )

    client = AuthenticatedClient(base_url=params["NILE_URL"], token=token.token)

    print(f"\n{ARROW_RIGHT} Logged into Nile as user {email}")
    print(f"export NILE_ACCESS_TOKEN={token.token}")

    # Check if org exists, create if not
    org_name = "Colton Labs"
    organizations = list_organizations.sync(client=client, workspace=params["NILE_WORKSPACE"])
    organization = next(
        (each for each in organizations if each.name == org_name),
        None,
    )
    if organization is not None:
        print(f"{GOOD} Organization {organization.name!r} exists with id {organization.id}")
    else:
        organization = create_organization.sync(
            workspace=params["NILE_WORKSPACE"],
            client=client,
            json_body=CreateOrganizationRequest(name=org_name),
        )
        print(f"{GOOD} Created new org {organization.name!r} with orgID {organization.id}")


if __name__ == "__main__":
    run()
